homeassistant/components/telegram_bot/__init__.py

```diff
@@ -579,9 +583,13 @@ def _make_row_inline_keyboard(row_keyboard):
             if ATTR_KEYBOARD in data:
                 keys = data.get(ATTR_KEYBOARD)
                 keys = keys if isinstance(keys, list) else [keys]
+                is_resize_keyboard = data[ATTR_RESIZE_KEYBOARD] if ATTR_RESIZE_KEYBOARD in data else False
+                is_one_time_keyboard = data[ATTR_ONE_TIME_KEYBOARD] if ATTR_ONE_TIME_KEYBOARD in data else False
                 if keys:
                     params[ATTR_REPLYMARKUP] = ReplyKeyboardMarkup(
-                        [[key.strip() for key in row.split(",")] for row in keys]
+                        [[key.strip() for key in row.split(",")] for row in keys],
+                        resize_keyboard = is_resize_keyboard,
+                        one_time_keyboard = is_one_time_keyboard
```

Comment:
Minor nit: `is_resize_keyboard` and `is_one_time_keyboard` are not used unless `keys` is non null.  So it probably better to just fetch them directly where they're assigned instead of pre-loading them outside the `if keys:` statement.

```suggestion
                        [[key.strip() for key in row.split(",")] for row in keys],
                        resize_keyboard = data[ATTR_RESIZE_KEYBOARD] if ATTR_RESIZE_KEYBOARD in data else False,
                        one_time_keyboard = data[ATTR_ONE_TIME_KEYBOARD] if ATTR_ONE_TIME_KEYBOARD in data else False
```

Reply:
At minimum, just move them just inside the if statement if there's some other reason these are separated out into their own staments.

Reply:
Ok, thanks!